mbtools/validate_mbz_html.py

```diff
@@ -0,0 +1,118 @@
+
+VALID_PREFIXES = ["https://s3.amazonaws.com/im-ims-export/",
+                  "https://k12.openstax.org/contents/raise"]
+
+
+class Violation:
+    def __init__(self, html_string, issue, location, details=None):
+        self.html = html_string
+        self.issue = issue
+        self.location = location
+        self.details = details
+
+    def tostring(self):
+        return f'{self.issue}\n At Location: {self.location}'
+
+    def toDict(self):
+        dict = {"issue": self.issue,
+                "location": self.location}
+        return dict
+
+
+def validate_mbz(mbz_path):
+
+    html_elements = utils.parse_backup_elements(mbz_path)
+
+    violations = []
+    violations.extend(find_style_violations(html_elements))
+    violations.extend(find_source_violations(html_elements))
+    violations.extend(find_tag_violations(html_elements))
+
+    return violations
+
+
+def find_style_violations(html_elements):
+    violations = []
+    for elem in html_elements:
+        attributes = elem.get_attribute_names()
+        if "style" in attributes:
+            violations.append(Violation(elem.tostring(),
+                                        STYLE_VIOLATION,
+                                        elem.location))
+    return violations
+
+
+def find_tag_violations(html_elements):
+    violations = []
+    for elem in html_elements:
+        hits = elem.get_child_elements("script")
+        if len(hits) > 0:
+            for hit in hits:
+                violations.append(Violation(elem.tostring(),
+                                            SCRIPT_VIOLATION,
+                                            elem.location,
+                                            hit))
+        hits = elem.get_child_elements("iframe")
+        if len(hits) > 0:
+            for hit in hits:
+                violations.append(Violation(elem.tostring(),
+                                            IFRAME_VIOLATION,
+                                            elem.location,
+                                            hit))
+    return violations
+
+
+def find_source_violations(html_elements):
+    violations = []
+    for elem in html_elements:
+        links = elem.get_attribute_values('src')
+        for link in links:
+            if len([prefix for prefix in VALID_PREFIXES if(prefix in link)]) \
+                    > 0:    # check if link contains a valid prefix
+                continue
+            elif "@@PLUGINFILE@@" in link:
+                violations.append(Violation(elem.tostring(),
+                                            MOODLE_VIOLATION,
+                                            elem.location,
+                                            link))
+            else:
+                violations.append(Violation(elem.tostring(),
+                                            SOURCE_VIOLATION,
+                                            elem.location,
+                                            link))
```

Comment:
It may be nice to include the actual offending prefix in the Violation output

Reply:
Can do. I think even with the restructure this is a quick fix